os/arch/arm/src/armv7-m/up_doirq.c

```diff
@@ -76,6 +76,7 @@
 #ifdef CONFIG_ARCH_NESTED_INTERRUPT
 extern uint32_t g_nestlevel; /* Initial top of interrupt stack */
 #endif
+int g_irq_nums[2] = {0}; /* Array to store the last two interrupt numbers */
```

Comment:
ARMV7 supports High Priority interrupts, so nesting can happen in v7 too. So I think we can add nesting check logic here similar to ARMV8

Reply:
Added the nested interrupt code to ARMV7M as well.